unison-cli/src/Unison/CommandLine/Completion.hs

```diff
@@ -184,24 +182,9 @@ completeWithinNamespace compTypes query codebase currentPath = do
 
     -- Regrettably there'shqFromNamedV2Referencenot a great spot to combinators for V2 references and shorthashes right now.
     hqFromNamedV2Referent :: Int -> V2Branch.NameSegment -> Referent.Referent -> HQ'.HashQualified V2Branch.NameSegment
-    hqFromNamedV2Referent hashLen n r = HQ'.HashQualified n (SH.take hashLen $ v2ReferentToShortHash r)
+    hqFromNamedV2Referent hashLen n r = HQ'.HashQualified n (Cv.referent2toshorthash1 (Just hashLen) r)
```

Comment:
Same distinction for `HashQualified` maybe; whether it's input from the user (it's whatever they typed, you get what you get) or output (it could be a name and/or `Hash`, and rendered however at the end of the world)